2364/2/parser.rb

```diff
@@ -0,0 +1,64 @@
+require 'russian_obscenity'
+require 'terminal-table'
+load 'participant.rb'
+load 'modules.rb'
+
+# Class parse for parsing data
+class Parser
+  PATH = File.expand_path('.').freeze
+  attr_reader :participants, :methods
+  def initialize(value, name)
+    @participants = []
+    @methods = create_methods
+    create_participants(Dir[PATH + '/rap-battles/*'])
+    @participants.sort_by! { |participant| participant.participant_data[:bad_words] }
+    @participants.reverse!
+    if name
+      print_top_words(name, value)
+    else
+      print_obscene_participants(value)
+    end
+  end
+
+  private
+
+  def print_top_words(name, value)
+    answer = participants.find { |participant| FunctionalMethods.name_check(participant.name, name) }
+    if answer
+      answer.count_words(value)
+    else
+      unknown_output(name)
+    end
+  end
+
+  def unknown_output(name)
+    puts "Рэпер #{name} не известен мне. Зато мне известны:"
+    participants[0...3].each { |raper| puts raper.name }
+  end
+
+  def print_obscene_participants(value)
+    rows = []
+    @participants = @participants[0..value - 1] if value
+    @participants.each { |participant| rows << participant.to_str }
+    puts Terminal::Table.new rows: rows
```

Comment:
don't mix parser and presenter classes, you should follow single responsibility principals here